src/client.rs

```diff
@@ -885,7 +878,7 @@ impl Client {
             // TODO: get rid of this clone
             let current = current.clone();
             // TODO: and this one too
-            let mut c = c.clone();
+            let c = c.clone();
```

Comment:
This clone _may_ no longer be necessary.

Reply:
Apparently it still is, due to lifetimes